From 2b999d8ed51e10af1cab13b494fbf34a1ee6e0e0 Mon Sep 17 00:00:00 2001
From: Alan Tull <atull@altera.com>
Date: Tue, 5 Feb 2013 16:32:31 -0600
Subject: [PATCH 079/248] FogBugz #117870: enable rtc

Enable ds1339 I2C rtc in defconfig and device tree for socfpga
cyclone5 board.

[Original patch taken from
git://git.rocketboards.org/linux-socfpga.git socfpga-3.14]

Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
Signed-off-by: Alan Tull <atull@altera.com>
---
 arch/arm/boot/dts/socfpga_cyclone5.dtsi | 5 +++++
 arch/arm/configs/socfpga_defconfig      | 6 ++++++
 2 files changed, 11 insertions(+)

diff --git a/arch/arm/boot/dts/socfpga_cyclone5.dtsi b/arch/arm/boot/dts/socfpga_cyclone5.dtsi
index aa319e9..3dc48c4 100644
--- a/arch/arm/boot/dts/socfpga_cyclone5.dtsi
+++ b/arch/arm/boot/dts/socfpga_cyclone5.dtsi
@@ -224,4 +224,9 @@
 		reg = <0x51>;
 		pagesize = <32>;
 	};
+
+	rtc@68 {
+		compatible = "dallas,ds1339";
+		reg = <0x68>;
+	};
 };
diff --git a/arch/arm/configs/socfpga_defconfig b/arch/arm/configs/socfpga_defconfig
index bfed41b..e9241e3 100644
--- a/arch/arm/configs/socfpga_defconfig
+++ b/arch/arm/configs/socfpga_defconfig
@@ -138,3 +138,9 @@ CONFIG_LEDS_TRIGGER_TIMER=y
 CONFIG_LEDS_TRIGGER_CPU=y
 CONFIG_FPGA=y
 CONFIG_FPGA_MGR_ALTERA=y
+CONFIG_RTC_LIB=y
+CONFIG_RTC_CLASS=y
+CONFIG_RTC_INTF_SYSFS=y
+CONFIG_RTC_INTF_PROC=y
+CONFIG_RTC_INTF_DEV=y
+CONFIG_RTC_DRV_DS1307=y
-- 
1.9.1

